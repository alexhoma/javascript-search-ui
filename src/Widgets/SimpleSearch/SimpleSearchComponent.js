/**
 * @jsx h
 */

import { h, Component } from 'preact';
import {simpleSearchAction} from "./simpleSearchActions";

/**
 * SimpleSearch Component
 */
class SimpleSearchComponent extends Component {
    shouldComponentUpdate() {
        return false;
    }

    handleSearch = (e) => {
        const {
            startSearchOn,
            environmentId,
            currentQuery,
            client
        } = this.props;

        /**
         * Search when string is bigger than {startSearchOn}
         */
        if (e.target.value.length < startSearchOn) return;

        /**
         * Dispatch input search action
         */
        simpleSearchAction(
            {   // queryOptions
                queryText: e.target.value
            },
            {   // appOptions
                environmentId,
                currentQuery,
                client
            }
        )
    };

    render() {
        const {
            placeholder,
            autofocus,
            classNames: {
                container: containerClassName,
                input: inputClassName
            }
        } = this.props;

        return (
            <div className={`asui-simpleSearch ${containerClassName}`}>
                <input
                    type='text'
                    className={`asui-simpleSearch--input ${inputClassName}`}
                    autofocus={autofocus}
                    placeholder={placeholder}
                    onInput={this.handleSearch}
                />
            </div>
        );
    }
}

SimpleSearchComponent.defaultProps = {
    placeholder: '',
    autofocus: false,
    startSearchOn: 0,
    classNames: {
        container: '',
        input: ''
    }
};

export default SimpleSearchComponent;