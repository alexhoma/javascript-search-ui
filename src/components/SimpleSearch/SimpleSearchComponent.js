/**
 * @jsx h
 */

import { h, Component } from 'preact';
import {simpleSearchAction} from "./simpleSearchActions";
import Template from "../Template";

/**
 * SimpleSearch Component
 */
class SimpleSearchComponent extends Component {

    handleSearch = (e) => {
        const {
            startSearchOn,
            environmentId,
            currentQuery,
            client
        } = this.props;

        /**
         * Search when string is bigger than {startSearchOn}
         */
        if (e.target.value.length < startSearchOn) return;

        /**
         * Dispatch input search action
         */
        simpleSearchAction(
            {
                queryText: e.target.value
            },
            {
                environmentId,
                currentQuery,
                client
            }
        )
    };

    clearSearch = () => {
        const {
            environmentId,
            currentQuery,
            client
        } = this.props;

        simpleSearchAction(
            {
                queryText: ''
            },
            {
                environmentId,
                currentQuery,
                client
            }
        )
    };

    render() {
        const {
            placeholder,
            autofocus,
            clearSearch,
            withContainer,
            classNames: {
                container: containerClassName,
                input: inputClassName,
                clearSearch: clearSearchClassName
            },
            template: {
                clearSearch: clearSearchTemplate
            },
            currentQuery: {
                q: currentQueryText
            },
            htmlNodeInheritProps
        } = this.props;

        let searchInput = (<input
            type='text'
            className={`as-simpleSearch__input ${inputClassName}`}
            {...htmlNodeInheritProps}
            autofocus={autofocus}
            placeholder={placeholder}
            onInput={this.handleSearch}
            value={currentQueryText}
            autocomplete="off"
        />);

        if (withContainer) {
            return (
                <div className={`as-simpleSearch ${containerClassName}`}>
                    {searchInput}

                    {(clearSearch && currentQueryText.length !== 0)
                        ? (
                            <div
                                className={`as-simpleSearch__clearSearch ${clearSearchClassName}`}
                                onClick={this.clearSearch}
                            >
                                <Template template={clearSearchTemplate}/>
                            </div>
                        ) : null
                    }
                </div>
            );
        }

        return searchInput;
    }
}

SimpleSearchComponent.defaultProps = {
    placeholder: '',
    autofocus: false,
    startSearchOn: 0,
    clearSearch: true,
    withContainer: true,
    classNames: {
        container: '',
        input: '',
        clearSearch: ''
    },
    template: {
        clearSearch: 'x'
    }
};

export default SimpleSearchComponent;