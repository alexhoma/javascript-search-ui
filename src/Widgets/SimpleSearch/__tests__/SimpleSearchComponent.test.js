/** @jsx h */
import {h} from 'preact';
import {shallow} from 'preact-render-spy';
import SimpleSearchComponent from './../SimpleSearchComponent';

describe('<SimpleSearchComponent />',() => {
    it('should render with default props', () => {
        const tree = shallow(<SimpleSearchComponent
            target={'#some-tagret'}
        />);

        expect(tree).toMatchSnapshot();
    });

    it('should render with custom props', () => {
        /**
         * Full custom props available object
         */
        const customProps = {
            classNames: {
                container: 'custom-container-class'
            }
        }

        const tree = shallow(<SimpleSearchComponent
            target={'#some-tagret'}
            {...customProps}
        />);

        expect(tree).toMatchSnapshot();
    });
});