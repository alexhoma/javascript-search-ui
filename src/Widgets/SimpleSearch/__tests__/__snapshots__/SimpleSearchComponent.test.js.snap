// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleSearchComponent /> should render with custom props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-simpleSearch custom-container-class">
  <input
    type="text"
    class="asui-simpleSearch--input undefined"
    autofocus={false}
    placeholder=""
    onInput={[Function anonymous]}
   />
</div>

`;

exports[`<SimpleSearchComponent /> should render with default props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-simpleSearch ">
  <input
    type="text"
    class="asui-simpleSearch--input "
    autofocus={false}
    placeholder=""
    onInput={[Function anonymous]}
   />
</div>

`;
